movie/src/components/Card/index.jsx

```diff
@@ -2,29 +2,71 @@ import "./Card.css";
 import Grid from "@mui/material/Grid";
 import { Item } from "semantic-ui-react";
 import DetailIcon from "../DetailIcon";
+import months from "../../data/Month.json";
+import { useState } from "react";
 const IMAGE_URL =
   "https://images.unsplash.com/photo-1503023345310-bd7c1de61c7d?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwcm9maWxlLXBhZ2V8N3x8fGVufDB8fHx8&w=1000&q=80";
 
 const Card = ({ movies }) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const newMovies = [...movies];
+  const [idCardClicked, setIdCardClicked] = useState(0);
+
+  const handleClick = (event, id) => {
+    setAnchorEl(event.currentTarget);
+    newMovies.forEach((mo) => {
+      if (mo.id === id) {
+        setIdCardClicked(mo.id);
+      }
+    });
+  };
+
+  const handleClose = () => {
+    setAnchorEl(false);
+    setIdCardClicked(0);
+  };
+
+  const isCheckOnCard = (idCard, idClick) =>
+    idCard === idClick ? "blur" : "none";
+
+  // Set date format
+  const formatDate = (newDate) => {
+    const d = new Date(newDate);
+    const year = d.getFullYear();
+    const date = d.getDate();
+    const monthIndex = d.getMonth();
+    const monthName = months[monthIndex];
+    const formatted = `${monthName} ${date} ${","} ${year}`;
+    return formatted.toString();
```

Comment:
1. too much `const`. Please try to minimize it. 
2. `const formatted = `${monthName} ${date} ${","} ${year}`;` is already a `string`...? why do we need the `.toString()` at the end?
3. please use `useMemo` for this function https://react.dev/reference/react/useMemo

Reply:
yep, I got it.
useMemo work unless value in this function have been change.But, It doesn't support parameter.
useCallback work unless value in this function have been changed. It supports parameber.